core/modules/savers/upload.js

```diff
@@ -28,10 +28,22 @@ UploadSaver.prototype.save = function(text,method,callback) {
 		password = $tw.utils.getPassword("upload"),
 		uploadDir = this.wiki.getTextReference("$:/UploadDir") || ".",
 		uploadFilename = this.wiki.getTextReference("$:/UploadFilename") || "index.html",
+		uploadWithUrlOnly = this.wiki.getTextReference("$:/UploadWithUrlOnly") || "no",
 		url = this.wiki.getTextReference("$:/UploadURL");
 	// Bail out if we don't have the bits we need
-	if(!username || username.toString().trim() === "" || !password || password.toString().trim() === "") {
-		return false;
+	if (uploadWithUrlOnly === "yes") {
+		// The url is good enough. No need for a username and password.
+		// Assume the server uses some other kind of auth mechanism.
```

Comment:
hmmmm, I'm OK, that there is an option, that upload is possible, if there is an other auth mechanism in place. But this setting seems to be a system tiddler. Wouldn't it be better, if the option would be a "startup parameter" for the server itself, like `--listen xxx` eg: `--allow-upload-with-url-only=yes` or something similar?

Reply:
By "server" here I mean something like Tiddlyspot. Imagine something like Tiddlyspot where the user creates a site, the empty TW file is created with $:/UploadURL pre-populated, then the user visits the new site and clicks save. Right now, it won't work unless the username and password are non-empty. The username could be pre-populated also, but I don't know how to pre-populate an upload password. And since they will be ignored, it seems preferrable not to require them.

Reply:
In the long term I'm thinking I could create a dedicated saver, but in the short term, this is a small patch that makes the existing upload.js work smoothly for this use case.